### PR TITLE
Update 02-server-actions-and-mutations.mdx

### PR DESCRIPTION
**Pull request by aktoriukas**
### Improving Documentation

mistake in the docs.

action should not be apply on the button.
only formAction or onClick is allow.

### CODE REVIEW

**Comment by samcx** [comment]
@aktoriukas Thanks for the catch! Let's use `onClick` instead here.

**Comment by aktoriukas** [comment]
ok @samcx ,

adding `onClick` instead of the `formAction`.

also adding `'use client'` as to use `onClick` component it has to be client component.

**Review by samcx — APPROVED** [review]
<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Comment by hnordt** [comment]
> @aktoriukas Thanks for the catch! Let's use `onClick` instead here.

@samcx just for curiosity and learning, why did you suggest to use `onClick` instead of `formAction` there?

File: docs/02-app/01-building-your-application/02-data-fetching/02-server-actions-and-mutations.mdx [hunk 1]
```diff
@@ -928,11 +929,12 @@ export default function Page() {
     ...
   }
 
-  return <button action={publish}>Publish</button>;
+  return <button onClick={publish}>Publish</button>;
```
Comment (A7med3bdulBaset) [inline docs/02-app/01-building-your-application/02-data-fetching/02-server-actions-and-mutations.mdx:932]: @samcx 
`onClick` pass `MouseEvent` not `FormData`. `formAction` was correct

Comment (samcx) [reply docs/02-app/01-building-your-application/02-data-fetching/02-server-actions-and-mutations.mdx:932]: @A7med3bdulBaset Thanks for the correction, will update!

**Pull request merged**